zerver/views/message_fetch.py

```diff
@@ -753,16 +787,26 @@ def exclude_muting_conditions(
 
     return conditions
 
-def join_with_usermessage(user_profile: UserProfile,
-                          query: Select):
-    
-    query = select([column("message_id")],
-                        and_(column("user_profile_id") == literal(user_profile.id),column("message_id").in_(query)),
-                        join(table("zerver_usermessage"), table("zerver_message"),
-                             literal_column("zerver_usermessage.message_id", Integer) ==
-                             literal_column("zerver_message.id", Integer)))
+
+def join_with_usermessage(user_profile: Optional[UserProfile], query: Select) -> Select:
+
+    assert user_profile is not None
+
+    query = select(
+        [column("message_id")],
+        and_(
+            column("user_profile_id") == literal(user_profile.id), column("message_id").in_(query)
+        ),
+        join(
+            table("zerver_usermessage"),
+            table("zerver_message"),
+            literal_column("zerver_usermessage.message_id", Integer)
+            == literal_column("zerver_message.id", Integer),
+        ),
+    )
     return query
```

Comment:
This looks to reformat this function; I suspect you want this hunk to be in an earlier commit (maybe coming from your doing the rebase across the Black transition).  You should be able to just run Black in the commit that extracts this function and amend that commit, and have Git automatically drop this block from the wrong commit.

Check out our [GitHub guide](https://zulip.readthedocs.io/en/latest/git-guide.html) and [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/version-control.html) for more advice on Git rebases.